File: QAClient/src/app/models/productReview.ts
import { Product } from "./product";
import { User } from "./user";

export interface ProductReview {
    prodRevId: number;
    productId: number;
    userId: string;
    comment: string;
    rating: number;
    dateReviewed: Date;

    product?: Product;
    user?: User;
}

File: QAAPI/Domain/Models/ProductReview.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Models
{
    public class ProductReview
    {
        [Key]
        public int ProdRevId { get; set; }

        [ForeignKey("ProductId")]
        public int ProductId { get; set; }

        [ForeignKey("UserId")]
        public string UserId { get; set; }
        public string Comment { get; set; }
        public int Rating { get; set; }
        public DateTime DateReviewed { get; set; }

        public Product Product { get; set; } // Many-to-One relationship
        public IdentityUser User { get; set; }
    }
}
